javascript/javascript1/week3/index.js

```diff
@@ -0,0 +1,205 @@
+    return percentage;
+
+}
+const firstSeries = logOutSeriesText(totalTimeForFirstSeries, myLifeSpan);
+const secondSeries = logOutSeriesText(totalTimeForSecondSeries, myLifeSpan);
+const thirdSeries = logOutSeriesText(totalTimeForThirdSeries, myLifeSpan);
+const inTotal = logOutSeriesText(inTotalTimeForSeries, myLifeSpan)
+console.log(`Game of thrones took${firstSeries}% of my life`);
+console.log(`Sopranos took${secondSeries}% of my life`);
+console.log(`The Wire took${thirdSeries}% of my life`);
+console.log(`In total that is${inTotal}% of my life`);
+
+// Step3: Note taking app!
+
+//3.1 save a note
+const notes = [];
+function saveNote(content, id){
+    let note = {
+        content: content,
+        id: id
+    };
+    return notes.push(note);
+}
+saveNote("Pickup groceries", 1);
+saveNote(" Do laundry", 2);
+console.log(notes);
+
+// 3.2 Get a note
+
+function getNote(id){
+    for(let i = 0; i<notes.length; i++){
+        if(id === isNaN() ||id ==='' ){
+            return  'please write a number in ID'
+        }
+        else if(id === (notes[i].id)){
+            return notes[i];
+        }
+        
+    }
+
+
+}
+const firstNote = getNote(1);
+console.log(firstNote);
+
+
+//3.3 log out notes
+
+function logOutNotesFormatted(){
+    for(i=0; i<notes.length; i++){
+        console.log(`The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}`);
+        
+    }
+}
+logOutNotesFormatted();
+
+// Adding a activity and Usage Limit
+
+const activities = [];
+
+function addActivity(date, activity, duration){
+    let addedActivities = {date: new Date(), activity: activity, duration: duration};
+    if (typeof activity === "string" && !isNaN(duration && activity!="")){
+        return activities.push(addedActivities);
+    }
+    else{
+        return console.log("activity should be string and duration is number");
+    }
+    
+    /*activities.date = new Date();
+    activities.activity = activity;
+    activities.duration = duration;
+    activities.push({date, activity, duration});*/
+}
+
+addActivity("", "Javascript function", 30);
+addActivity("", "Fundamental of JS", 50);
+addActivity("", "Why Javascript1", 80);
+addActivity("", "Why Javascript", 20); // this is not calculate limit for only 3 activities.
+console.log(activities);
+
+
+// Show my status and Usage Limit
+
+
+
+function showStatus(){
+    let sum = 0;
+    let count = 0;
+    activities.length = 3; // usage limit so user could not get more than 3 activities.
+    for(i=0; i<activities.length; i++){
+        sum= sum + activities[i].duration;
+        count ++;
+    }
+    console.log(`You have added ${count} activities. The amount to ${sum} min of usage`);
+    console.log(`You have reached your limit, no more smartphoning for you`);
+    }
+showStatus();
+
+//Optional
+// we don't need to specify date - done in addActivity function
+// calculate the activity a user has spent the most time on.
+
+function mostTimeSpent(){
```

Comment:
It's useful that we can get the most time spent. It would be nice to know what this time was spent on :)

Reply:
I calculate most time spent of activities  but I couldnot display the activity name of that most pent time